docs/upgrade.md

```diff
@@ -88,6 +88,28 @@ process, for example:
     dpkg -i matrix-synapse-py3_1.3.0+stretch1_amd64.deb
     ```
 
+# Upgrading to v1.83.0
+
+## Worker Yaml setting deprecations
```

Comment:
I'd probably rejig the title and try and make it clearer that the 3 settings not only can be removed, but *should* be removed because they are now deprecated (though it's also worth being fair that there is still backwards compat for now).

Reply:
So...
```yaml
## Worker YAML setting removal and deprecations
```
...instead? I guess I don't follow

Reply:
I guess your PR title has it more or less spot on... how about:

`## Deprecation of `worker_replication_*` configuration options`

Then just make the upgrade notes as clear as possible about what you need to migrate (e.g. a before and after example)

Reply:
Gave this a shot in [e3f7936](https://github.com/matrix-org/synapse/pull/15491/commits/e3f7936f1d2a54a9f242c035262399d626286b36)